app/controllers/weekly_friends_controller.rb

```diff
@@ -1,14 +1,16 @@
 class WeeklyFriendsController < ApplicationController
   def show
     user = User.find(params["id"])
-    weekly_activity = WeeklyActivity.active.includes_user(user.id)
+    active = WeeklyActivity.active
+
+    if active.where(user_1_id: user.id).exists?
+      render json: active.find_by(user_1_id: user.id).user_2
+
+    elsif active.where(user_2_id: user.id).exists?
+      render json: active.find_by(user_2_id: user.id).user_1
 
-    if weekly_activity.nil?
-      render json: user
-    elsif user.id == weekly_activity.pluck(:user_1_id)
-      render json: User.where(id: weekly_activity.pluck(:user_2_id))
     else
-      render json: User.where(id: weekly_activity.pluck(:user_1_id))
+      render json: user
```

Comment:
What you're saying here is that if the user doesn't have an assigned friend he is connected to himself? Why not have some sort of response that explains what happens?

```suggestion
      render json: nil
```

or could even argue that this would also work as a 404:

```suggestion
      render json: {error: "not_found"}, status: 404
```

Reply:
That would also change the frontend but yes we can do that.